controllers/spec/common.go

```diff
@@ -423,7 +423,10 @@ func generateResource(resources corev1.ResourceList) *proto.Resources {
 	}
 }
 
-func getUserConfig(configs map[string]string) string {
+func getUserConfig(configs *v1alpha1.Config) string {
+	if configs == nil {
+		return ""
```

Comment:
"" might not be a valid json, will that be a problem?

Reply:
the `userConfig` in pulsar functions' proto `FunctionDetails` is defined as string, and the default value is `""`, so `""` would be fine if there is no `userConfig` provided.